keps/sig-node/5283-dra-resourceslice-status-device-health/README.md

```diff
@@ -0,0 +1,1052 @@
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+With Dynamic Resource Allocation (DRA), devices that can be allocated to Pods
+are advertised in ResourceSlices. Device entries in a ResourceSlice describe
+properties like identifiers, capacity, and firmware or kernel driver versions.
+
+Sometimes devices fail and an automated agent or cluster administrator must take
+some action to restore the device to a healthy state. This KEP defines a
+standard way to determine whether or not a device is considered healthy.
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Since the DRA APIs currently do not expose a standard way to classify device
+health, cluster administrators are tasked with finding other means of doing that
+for each kind of device that is defined in a ResourceSlice in order to identify
+and restore unhealthy devices back to their full functionality.
+
+When unhealthy devices are no longer able to support new or existing workloads,
+DRA drivers and cluster administrators already have the option to
+[taint](https://kep.k8s.io/5055) the devices, which always carries a
+`NoSchedule` or `NoExecute` side effect. A standard representation of device
+health in the ResourceSlice API that is strictly informational enables the
+widest variety of potential integrations (e.g. Node Problem Detector, custom
+controllers, dashboards) which may implement custom mitigation strategies that
+conflict with other side effects.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Expose well-formed data in the ResourceSlice API representing the real-time
+  health of the devices a ResourceSlice describes.
```

Comment:
```suggestion
- Expose well-formed data in the Kubernetes API representing the real-time
  health of devices that the ResourceSlices describe.
```

This is a bit more neutral regarding how this will be achieved. Depends of course on whether there are use cases that can only be satisfied if the data really is in the ResourceSlice.